### DISCUSSION TITLE
Only load specified fields

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#194] [Q&A]
An exception will occur when sqlalchemy2 uses load_only to load only specified fields.

Are there any working examples? Thanks

**Comment by wu-clan** [comment]
Sorry, only used load_only in 1.4.x, but that code has been overwritten!

Maybe in the docs:
https://docs.sqlalchemy.org/en/20/orm/queryguide/columns.html#using-load-only-to-reduce-loaded-columns

It's worth noting: don't use a string for the load_only field, but a table field

**Comment by downdawn** [comment]
Example
```
    async def get(self, db: AsyncSession, pk: int) -> Api | None:
        api = await db.execute(
            select(self.model).options(load_only(self.model.path, self.model.method)).where(self.model.id == pk))
        return api.scalars().first()
```
Throw an exception
```
sqlalchemy.orm.exc.DetachedInstanceError: Parent instance <Api at 0x2465ce846d0> is not bound to a Session; deferred load operation of attribute 'created_time' cannot proceed (Background on this error at: https://sqlalche.me/e/20/bhk3)
```

**Reply by downdawn** [reply]
![image](https://github.com/fastapi-practices/fastapi_best_architecture/assets/41266749/6fa30845-8077-442f-be89-ec6544d93835)
...

**Comment by wu-clan** [comment]
This is because the entire instance is returned in the query, and the entire instance is serialized before it is finally returned, and because only some of the fields are loaded, other fields will raise an error when serialized

Solution: Use select_to_dict or select_to_json can to exclude the wrong fields.

**Reply by downdawn** [reply]
OK
But if the entire instance is returned in the query, the effect is limited.

**Reply by wu-clan** [reply]
This is related to the sqla working mechanism. Maybe you can ask them questions on github, and their responses are very fast.
